content/general/concepts/what-is-functional-programming/what-is-functional-programming.md

```diff
@@ -0,0 +1,53 @@
+---
+Title: 'Functional Programming'
+Description: 'Encourages programming with pure functions and evaluating expressions instead of statements.'
+Subjects:
+  - 'Code foundations'
+  - 'Computer Science'
+  - 'Web Development'
+Tags:
+  - 'Functions'
+  - 'Recursion'
+CatalogContent:
+  - 'learn-to-code'
+  - 'paths/code-foundations'
+---
+
+_Functional Programming_ (FP) is a declarative software development paradigm which encourages constructing programs by composing pure functions and evaluating expressions instead of statements. The functions are isolated and independent of the state of the application.
+
+In FP, dependencies such as variables or objects needed for a function, are always declared explicitly by passing them into the function as arguments. Any operation inside the function is therefore bound to the arguments passed, instead of depending on global objects or variables. This makes the function:
+
+- Suitable for testing and debugging.
+- Predictable, reusable, and stable when refactoring code.
+
+FP helps make code more modular, readable and easier to reason about in general.
+
+## Core Principles of FP
+
```

Comment:
Could we add a small intro here about the core principles of FP before going into each one?